pcb/rules/F6_2.py

```diff
@@ -24,19 +25,31 @@ def check(self):
             # Ignore non-smd parts
             return False
 
-        center = module.padMiddlePosition()
+        center_pads = module.padMiddlePosition()
+        center_fab  = module.geometricBoundingBox("F.Fab").center
+
+        if center_pads['x'] != center_fab['x'] or center_pads['y'] != center_fab['y']:
+            self.warning("Footprint centers of pads and F.Fab do not match")
+            self.warningExtra("Footprint center calculated as Pad: ({xp},{yp})mm F.Fab: ({xf}, {yf})mm".format(
+                xp = round(center_pads['x'], 5),
+                yp = round(center_pads['y'], 5),
+                xf = round(center_fab['x'], 5),
+                yf = round(center_fab['y'], 5)))
 
         err = False
 
         THRESHOLD = 0.001
-        x = center['x']
-        y = center['y']
+        # select the xy coordinates that are closest to the center
+        diff_pads = sqrt(center_pads['x']**2 + center_pads['y']**2)
+        diff_fab  = sqrt(center_fab['x']**2 +  center_fab['y']**2)
+        x = (center_pads['x'], center_fab['x'])[diff_pads > diff_fab]
+        y = (center_pads['y'], center_fab['y'])[diff_pads > diff_fab]
```

Comment:
What is the rationale behind this?

Reply:
Line 43 & 44 do calculate the euclidic distance of the pads & fab centers to the origin.
Then the one closes to the origin is selected. (45-46)

That the acutal idea of the patch. Look at the center of the F.Fab lines the one calculated from the Pads.
* If they are not the same, display a warning (line 31-37)
* If they are not the same, select the one that is closer to the footprint origin (line 43-36)
* Then check if the one selected above is to far from the origin (lines 48-52)

Reply:
Just a stylistic point: Python has a conditional operator `true_value if condition else false_value` that would be better to use than subscripting a tuple on lines 45-46.  But since there are two lines with the same condition, using a normal `if` block could be considered as the more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) option.

Reply:
tbh. I don't know why I have written it this way. After the question from poeschlr I had to look twice myself. This is a good indication that the code-quality is shitty.
A simple if-statement is way more readable. Will fix.